lib/plugins/setedit/syntax.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * setedit plugin, allows users to set his editor preference
+ * 
+ * @license GPL 2 (http://www.gnu.org/licenses/gpl.html)
+ *
+ * @author Yvonne Lu <yvonnel@leapinglaptop.com>
+ *
+ * 
+ */
+
+if(!defined('NL')) define('NL', "\n");
+if(!defined('DOKU_INC')) define('DOKU_INC', dirname(__FILE__) . '/../../');
+if(!defined('DOKU_PLUGIN')) define('DOKU_PLUGIN', DOKU_INC . 'lib/plugins/');
+require_once(DOKU_PLUGIN . 'syntax.php');
+require_once(DOKU_INC . 'inc/media.php');
+require_once(DOKU_INC . 'inc/auth.php');
+
+require_once(DOKU_INC . 'inc/infoutils.php');
+
+//define for debug
+define ('RUN_STATUS', 'SERVER');
+
+class syntax_plugin_setedit extends DokuWiki_Syntax_Plugin {
+
+    var $fh=NULL; //debug file handle
+    
+    function getInfo() {
+        return array(
+            'author' => 'Yvonne Lu',
+            'email' => 'yvonnel@leapinglaptop.com',
+            'date' => '2015-6-24',
+            'name' => 'setedit plugin',
+            'desc' => 'setedit plugin dumps browser cookie {setedit}',
+            'url' => '',
+        );
+    }
+
+    function getType() {
+        return 'substition';
+    }
+
+    function getSort() {
+        return 32;
+    }
+
+    function connectTo($mode) {
+        
+        $this->Lexer->addSpecialPattern('\{setedit\}', $mode, 'plugin_setedit');
+    }
+
+    function handle($match, $state, $pos, &$handler) {
+        
+    }
+
+    function render($mode, &$renderer, $data) {
+        global $ID;
+  
+        $this->showDebug('in render $mode='.$mode);
+        $sel_str= '<input type="radio" name="select_edit" id = "dw" value="dw" checked="checked">'.
+                            '<label for="dw">'.$this->getLang('dw_edit').'</label><br>'.
+                        '<input type="radio" name="select_edit" id = "ckg" value="ckg">'.
+                            '<label for="dw">'.$this->getLang('ckgedit').'</label><br>';
```

Comment:
This is just a suggestion. I find it easier to debug and maintain if I create a separate html file. The door43shared plugin has a function translateHtml that will insert the values from lang.php.